chapters/data-analysis.tex

```diff
@@ -1,292 +1,488 @@
+such as \texttt{highcharter}\sidenote{\url{http://jkunst.com/highcharter/}}, 
+\texttt{r2d3}\sidenote{\url{https://rstudio.github.io/r2d3/}}, 
+\texttt{leaflet}\sidenote{\url{https://rstudio.github.io/leaflet/}}, 
+and \texttt{plotly}\sidenote{\url{https://plot.ly/r/}}, to name a few.
+We have no intention of creating an exhaustive list, and this one is certainly missing very good references.
+But at least it is a place to start.
+
+\section{Exporting analysis outputs}
+
+% Exploratory analysis
+It's ok to not export each and every table and graph created during exploratory analysis. 
+Instead, we suggest running them into markdown files using RMarkdown or the different dynamic document options available in Stata. 
+This will allow you to update and present results quickly while maintaining a record of the different analysis tried. 
+% Final analysis
+Final analysis scripts, on the other hand, should export final outputs, which are ready to be included to a paper or report.
+No manual edits, including formatting, should be necessary after exporting final outputs. 
+Manual edits are difficult to replicate, and you will inevitably need to make changes to the outputs. 
+Automating them will save you time by the end of the process. 
+However, don't spend too much time formatting tables and graphs until you are ready to publish.\sidenote{For a more detailed discussion on this, including different ways to export tables from Stata, see \url{https://github.com/bbdaniels/stata-tables}}
+Polishing final outputs can be a time-consuming process, 
+and you want to it as few times as possible.
+
+We cannot stress this enough: don't ever set a workflow that requires copying and pasting results from the console.
+There are numerous commands to export outputs from both R and Stata.\sidenote{Some examples are \href{ http://repec.sowi.unibe.ch/stata/estout/}{\texttt{estout}}, \href{https://www.princeton.edu/~otorres/Outreg2.pdf}{\texttt{outreg2}}, 
+and \href{https://www.benjaminbdaniels.com/stata-code/outwrite/}{\texttt{outwrite}} in Stata, 
+and \href{https://cran.r-project.org/web/packages/stargazer/vignettes/stargazer.pdf}{\texttt{stargazer}}
+and \href{https://ggplot2.tidyverse.org/reference/ggsave.html}{\texttt{ggsave}} in R.}
+Save outputs in accessible and, whenever possible, lightweight formats.
+Accessible means that it's easy for other people to open them.
+In Stata, that would mean always using \texttt{graph export} to save images as \texttt{.jpg}, \texttt{.png}, \texttt{.pdf}, etc., 
+instead of \texttt{graph save}, which creates a \texttt{.gph} file that can only be opened through a Stata installation.
+Some publications require ``lossless'' TIFF of EPS files, which are created by specifying the desired extension.
+For tables, \texttt{.tex} is preferred. 
+Excel \texttt{.xlsx} and \texttt{.csv} files are also acceptable, 
+although if you are working on a large report they will become cumbersome to update after revisions.
```

Comment:
We can be more specifc it here. It is _not_ difficult to update .xlsx and .csv files. You just re-run the code. But what is difficult is to update those files and make sure the report is updated.

Reply:
> Excel \texttt{.xlsx} and \texttt{.csv} files are also acceptable, but require the extra step of copying the tables into the final output, so it can be cumbersome to ensure that your paper or report is always up-to-date.

(c80a095)

Reply:
And the workflow that we want to discourage for sure is exporting tables to excel and then copy-pasting those into formatted tables in word. that's common and super error-prone!

Reply:
> And the workflow that we want to discourage for sure is exporting tables to excel and then copy-pasting those into formatted tables in word. that's common and super error-prone!

Feeling seen 🥰

Reply:
edited a bit in f6c9917